sdgym/benchmark.py

```diff
@@ -630,10 +726,19 @@ def _run_job(args):
     )
 
     if synthesizer_path is not None:
-        synth_path = Path(synthesizer_path['synthesizer'])
-        root_path = synth_path.parents[2]
-        result_file = root_path / 'results.csv'
-        _safe_append(scores, result_file)
+        if s3_client:
+            bucket, keys = _parse_s3_paths(synthesizer_path)
+            synthesizer_path = keys['synthesizer']
+            parsed_url = urlparse(synthesizer_path)
+            key = parsed_url.path.lstrip('/')
+            root_folder = '/'.join(key.split('/')[:3])
+            results_key = f'{root_folder}/results.csv'
+            _upload_dataframe_to_s3(scores, s3_client, bucket, results_key, append=True)
+        else:
+            synth_path = Path(synthesizer_path['synthesizer'])
+            root_path = synth_path.parents[2]
+            result_file = root_path / 'results.csv'
+            _safe_append(scores, result_file)
```

Comment:
since there are some places that have these if else splits for local vs remote, should we make classes that represent the ResultsWriter, and then pass either a LocalResultsWriter or AWSResultsWriter to these functions? Then we don't need the if/else blocks. We can just do writer.update_results or whatever functions we pick